src/sage/combinat/crystals/generalized_young_walls.py

```diff
@@ -554,13 +554,10 @@ def weight(self, root_lattice=False):
             sage: x.weight(root_lattice=True)
             -2*alpha[0] - 3*alpha[1] - 5*alpha[2] - 3*alpha[3]
         """
-        W = []
         E = self.cartan_type().root_system().weight_lattice(extended=True)
         L = self.cartan_type().root_system().root_lattice()
         alpha = L.simple_roots()
-        for r in self.data:
-            for i in r:
-                W.append(-1*alpha[i])
+        W = [-1*alpha[i] for r in self.data for i in r]
```

Comment:
why using `-1*alpha[i]` and not using `-alpha[i]` ?

Reply:
just keeping what is written as there may be a good reason to use this

Reply:
You are right, it's safer.